defaults/main.yml

```diff
@@ -13,11 +13,7 @@ ha_cluster_start_on_boot: true
 
 ha_cluster_extra_packages: []
 
-ha_cluster_fence_agent_packages: "{{
-    ['fence-agents-all']
-    +
-    (['fence-virt'] if ansible_architecture == 'x86_64' else [])
-  }}"
+ha_cluster_fence_agent_packages: []
```

Comment:
```suggestion
ha_cluster_fence_agent_packages: "{{ __ha_cluster_fence_agent_packages }}"
```

Reply:
I don't think that is correct place, because it would expose private variable to potential user inputs.

I have added conditional into main yesterday to do switch, depending if users added something to it.
```
+
      ha_cluster_fence_agent_packages
        if ha_cluster_fence_agent_packages | length > 0
        else __ha_cluster_fence_agent_packages }}"
```

Reply:
> I don't think that is correct place, because it would expose private variable to potential user inputs.

I'm not sure what you mean.  I believe my proposal encapsulates the desired behavior - putting this in `defaults/main.yml`:
```yaml
ha_cluster_fence_agent_packages: "{{ __ha_cluster_fence_agent_packages }}"
```
This allows users to provide their own list of packages for `ha_cluster_fence_agent_packages` (which is the desired behavior - @tomjelinek please correct me if I'm wrong), and if the user does not specify `ha_cluster_fence_agent_packages`, it will be set to the default value `__ha_cluster_fence_agent_packages`.  And, since `__ha_cluster_fence_agent_packages` is defined for different values depending on the platform/version, we get the correct value for `ha_cluster_fence_agent_packages` for all platforms/versions.

> 
> I have added conditional into main yesterday to do switch, depending if users added something to it.
> 
> ```
> +
>       ha_cluster_fence_agent_packages
>         if ha_cluster_fence_agent_packages | length > 0
>         else __ha_cluster_fence_agent_packages }}"
> ```

I think this is not the "Ansible way" to do this.  I believe the correct way to do this is the way I outlined this above.

Reply:
@richm I have changed to in latest commit after conversation with @tomjelinek.

Reply:
Looks like the default variable was renamed to `__ha_cluster_fence_agent_packages`, so this should be
```yaml
ha_cluster_fence_agent_packages: "{{ __ha_cluster_fence_agent_packages_default }}"
```

Reply:
@richm Changes are completed and pushed.

Reply:
@richm this can be marked as resolved?